.github/actions/setup-from-previous/action.yaml

```diff
@@ -4,12 +4,6 @@ description: grabs artifact from a previous job and sets up the env for tests
 runs:
   using: composite
   steps:
-    # Checkout the repo and setup the tooling for this job
-    - name: Checkout
-      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
-      with:
-        fetch-depth: 0
-
```

Comment:
I noticed that every invocation of `setup-from-previous` also ran the `Checkout` step, so we were doing checkout twice in every use of this action

Reply:
All **Steps** in a **Job** run in the same container, and IIRC they start in the root of the repo unless specified. I verified that all invocations of `setup-from-previous` were ran _after_ a previous step ran `checkout`, so it should be good. There's nothing codifying that though, so we could actually remove those instances of checkout and just put it in this file instead, but files like `release.yaml` and such would still need to call `checkout` themselves since they don't use `setup-from-previous`

Reply:
> There's nothing codifying that though, so we could actually remove those instances of checkout and just put it in this file instead...

Fwiw you shouldn't even be able to call an action unless a checkout has occurred so I actually think this is guaranteed. You have to checkout the repo to be able to access a local action file, otherwise the action can't be found.

Reply:
@mjnagel ah great point, thanks. I either forgot or just didn't know that definitively:

- https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-pre-written-building-blocks-in-your-workflow#adding-an-action-from-the-same-repository